react/components/SKUSelector/index.js

```diff
@@ -14,9 +14,67 @@ import {
  * Display a list of SKU items of a product and its specifications.
  */
 class SKUSelectorContainer extends Component {
+
+  static defaultProps = {
+    alwaysShowSecondary: true,
+  }
+
+  state = { mainVariation: null, secondaryVariation: null }
+
+  buildVariations = (rawSkuSelected) => {
+    const skuSelected = rawSkuSelected && parseSku(rawSkuSelected)
+    const skuItems =
+      this.props.skuItems && this.props.skuItems.map(sku => parseSku(sku))
+    const itemId = skuSelected.itemId
+    const variations = skuSelected.variations
+
+    const name = getMainVariationName(variations)
+    const mainVariation = {
+      name,
+      value: this.props.skuSelected ? skuSelected[name] : null,
+      options: getVariationOptions(name, skuItems),
+    }
+
+    const secondaryVariation = { value: this.props.skuSelected ? itemId : null }
+
+    const filteredSkus = skuItems.filter(sku => sku[name] === skuSelected[name])
+
+    if (variations.length > 1) {
+      secondaryVariation.name = variations.find(variation => variation !== name)
+      secondaryVariation.options = getVariationOptions(
+        secondaryVariation.name,
+        filteredSkus
+      )
+    }
+
+    return { mainVariation, secondaryVariation }
+  }
+
+  componentDidMount() {
+    this.setState(this.buildVariations(this.props.skuSelected || this.props.skuItems[0]))
+  }
+
   handleSkuSelection = (isMainVariation, skuId) => {
+
+    const selectedSku = this.props.skuItems.find(({ itemId }) => itemId === skuId)
+    const variations = this.buildVariations(selectedSku)
```

Comment:
just keep an eye on this function call here, if you notice performance issues for complex skus